File: mclimate/src/main/java/com/mordrum/mclimate/MClimate.java
package com.mordrum.mclimate;

import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.SidedProxy;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;

@Mod(modid = MClimate.MOD_ID, name = MClimate.MOD_NAME, version = MClimate.MOD_VERSION)
public class MClimate {
    public static final String MOD_ID = "mclimate";
    public static final String MOD_NAME = "mClimate";
    public static final String MOD_VERSION = "1.0";

    @Mod.Instance(MOD_ID)
    private static MClimate instance;

    @SidedProxy(clientSide = "com.mordrum.mclimate.client.ClientProxy", serverSide = "com.mordrum.mclimate.server.ServerProxy")
    private static CommonProxy proxy;

    @Mod.EventHandler
    public void preInit(FMLPreInitializationEvent event) {
        proxy.onPreInit(event);
    }

    @Mod.EventHandler
    public void init(FMLInitializationEvent event) {
        proxy.onInit(event);
    }
}

File: mclimate/src/main/java/com/mordrum/mclimate/MClimate.kt
package com.mordrum.mclimate

import com.mordrum.mclimate.common.CommonProxy
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.SidedProxy
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent

@Mod(modid = MClimate.MOD_ID, name = MClimate.MOD_NAME, version = MClimate.MOD_VERSION)
class MClimate {

    @Mod.EventHandler
    fun preInit(event: FMLPreInitializationEvent) {
        proxy.onPreInit(event)
    }

    @Mod.EventHandler
    fun init(event: FMLInitializationEvent) {
        proxy.onInit(event)
    }

    companion object {
        const val MOD_ID = "mclimate"
        const val MOD_NAME = "mClimate"
        const val MOD_VERSION = "\$VERSION"

        @Mod.Instance(MOD_ID)
        private lateinit var instance: MClimate

        @SidedProxy(clientSide = "com.mordrum.mclimate.client.ClientProxy", serverSide = "com.mordrum.mclimate.server.ServerProxy")
        private lateinit var proxy: CommonProxy
    }
}
